src/UID2.Client/KeyContainer.cs

```diff
@@ -104,6 +110,11 @@ public bool TryGetMasterKey(DateTime now, out Key key)
             return TryGetKeysetActiveKey(_masterKeysetId, now, out key);
         }
 
+        public bool IsDomainNameForSite(int siteId, string domainName)
+        {
+            return _sites.TryGetValue(siteId, out var site) && site.HasDomainName(domainName);
```

Comment:
from UID2.Client.UID2Client.Decrypt in theory it can pass domainName = null into here, should we do an explicit `domainName != null` check at the front here (or in UID2Encryption.IsValidDomainNameForSite) in case for other reason site._domainNames has `null` value (which is theoretically possible?)? 

i dont think it would happen but just wanna be defensive

Reply:
i have added a if(domainName == null) check here